pkg/psmdb/service.go

```diff
@@ -185,45 +200,20 @@ func getIngressPoint(pod corev1.Pod, cl client.Client) (string, error) {
 }
 
 // GetReplsetAddrs returns a slice of replset host:port addresses
-func GetReplsetAddrs(cl client.Client, m *api.PerconaServerMongoDB, replset *api.ReplsetSpec, pods []corev1.Pod) ([]string, error) {
+func GetReplsetAddrs(m *api.PerconaServerMongoDB, replset *api.ReplsetSpec, pods []corev1.Pod) ([]string, error) {
 	addrs := make([]string, 0)
 
 	for _, pod := range pods {
-		host, err := MongoHost(cl, m, replset, pod)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get external hostname for pod %s", pod.Name)
-		}
+		host := MongoHost(m, replset, pod)
 		addrs = append(addrs, host)
 	}
 
 	return addrs, nil
 }
 
 // MongoHost returns the mongo host for given pod
-func MongoHost(cl client.Client, m *api.PerconaServerMongoDB, replset *api.ReplsetSpec, pod corev1.Pod) (string, error) {
-	if replset.Expose.Enabled {
```

Comment:
can we analyse ExternalDnsZone
1) if ExternalDnsZone set - return `fmt.Sprintf("%s.%s", podName, replset.Expose.ExternalDnsZone)`
2) if ExternalDnsZone = "" then `return getExtAddr(cl, m.Namespace, pod)` 
?

Reply:
That would mean that MongoHost would always return the service when enabled, either using the hostname or IP of the service, right?
Since MongoHost is used to get get the host name for cluster internal communication (i.e. operator to MongoDb) I think using the services complicates things here (communication would go from inside the cluster through the service just to be routed back to the Mongo pods). This would also require additional FW rules to allow the operator pod to connect to the Mongo pods through that service when using LoadBalancer.
Maybe I'm missing something?

Reply:
> That would mean that MongoHost would always return the service when enabled, either using the hostname or IP of the service, right?

Idea is to analyze ExternalDnsZone, if it is empty, that code will behave as before adding ExternalDnsZone feature. if it is set then use ExternalDnsZone way.

>  I think using the services complicates things here

yes, you right, but it is current functionality and we don't want to delete current feature.

Reply:
Alright, changed it back to the previous behaviour and added the requested change w.r.t. ExternalDnsZone.
I understand that you don't want to change the behaviour right now for compatibility reasons, but would it be an option to either change it at a later point or add an additional configuration that allows users to opt out of using external service for this?